Tests/xSharePoint/xSharePoint.xSPFarmSolution.Tests.ps1

```diff
@@ -0,0 +1,138 @@
+[CmdletBinding()]
+param(
+    [string] $SharePointCmdletModule = (Join-Path $PSScriptRoot "..\Stubs\SharePoint\15.0.4693.1000\Microsoft.SharePoint.PowerShell.psm1" -Resolve)
+)
+
+$ErrorActionPreference = 'stop'
+Set-StrictMode -Version latest
+
+$RepoRoot = (Resolve-Path $PSScriptRoot\..\..).Path
+$Global:CurrentSharePointStubModule = $SharePointCmdletModule 
+
+$ModuleName = "MSFT_xSPFarmSolution"
+Import-Module (Join-Path $RepoRoot "Modules\xSharePoint\DSCResources\$ModuleName\$ModuleName.psm1")
+
+Describe "xSPFarmSolution" {
+	
+	InModuleScope $ModuleName {
+	
+		$testParams = @{
+            Name = "MySolution.wsp"
+            LiteralPath = "MySolution.wsp"
+        }
+        Import-Module (Join-Path ((Resolve-Path $PSScriptRoot\..\..).Path) "Modules\xSharePoint")
+
+        Mock Invoke-xSharePointCommand { 
+            return Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Arguments -NoNewScope
+        }
+        
+        Import-Module $Global:CurrentSharePointStubModule -WarningAction SilentlyContinue
+
+		Context "When the solution does not exist,"{
```

Comment:
When we are creating tests and the contexts for them, we try to describe it as "the scenario is this, and should be this" - so for example:

 - The solution isn't installed, but should be
 - The solution isn't installed, and should not be

Then from there we would have "it" statements to test each of the three resources, so things like "it returns 0 from the get method", "it returns false from the test method", "the set method runs the install function" and things like that. It gives us a good structure to test for the potential states a system could be in, and validate the logic we have in each function. So are you able to have a look at the structure you have for the tests here? This first context for example you are essentially testing each property in its own test, and not really describing what the desired state of the context is either. Happy to help with it if that doesn't make sense though - let me know though.